adapters/adhese/adhese.go

```diff
@@ -0,0 +1,247 @@
+			var targetValueParsed = targetRawValKey.(string)
+			parametersAsString += targetValueParsed + ";"
+		}
+		parametersAsString = strings.TrimRight(parametersAsString, ";")
+	}
+
+	params := strings.Split(parametersAsString, "/")
+	sort.Strings(params)
+	return strings.Join(params, "/")
+
+}
+
+func extractGdprParameter(request *openrtb.BidRequest) string {
+	if request.User != nil {
+		var extUser openrtb_ext.ExtUser
+		if err := json.Unmarshal(request.User.Ext, &extUser); err == nil {
+			return "/xt" + extUser.Consent
+		}
+	}
+	return ""
+}
+
+func extractRefererParameter(request *openrtb.BidRequest) string {
+	if request.Site != nil && request.Site.Page != "" {
+		return "/xf" + url.QueryEscape(request.Site.Page)
+	}
+	return ""
+}
+
+func (a *AdheseAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	var err error
+
+	// If all the requests are invalid, Call to adaptor is skipped
+	if len(request.Imp) == 0 {
+		errs = append(errs, WrapError("Imp is empty"))
+		return nil, errs
+	}
+
+	var imp = &request.Imp[0]
+	var bidderExt adapters.ExtImpBidder
+
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		errs = append(errs, WrapError("Request could not be parsed as ExtImpBidder due to: "+err.Error()))
+		return nil, errs
+	}
+
+	var params openrtb_ext.ExtImpAdhese
+	if err := json.Unmarshal(bidderExt.Bidder, &params); err != nil {
+		errs = append(errs, WrapError("Request could not be parsed as ExtImpAdhese due to: "+err.Error()))
+		return nil, errs
+	}
+
+	// Compose url
+	endpointParams := macros.EndpointTemplateParams{Host: "ads-" + params.Account + ".adhese.com"}
+
+	host, err := macros.ResolveMacros(*&a.endpointTemplate, endpointParams)
+	if err != nil {
+		errs = append(errs, WrapError("Could not compose url from template and request account val: "+err.Error()))
+		return nil, errs
+	}
+	complete_url := fmt.Sprintf("%s%s%s%s%s",
+		host,
+		extractSlotParameter(params),
+		extractTargetParameters(params),
+		extractGdprParameter(request),
+		extractRefererParameter(request))
+
+	return []*adapters.RequestData{{
+		Method: "GET",
+		Uri:    complete_url,
+	}}, errs
+}
+
+func (a *AdheseAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{WrapError(fmt.Sprintf("Unexpected status code: %d.", response.StatusCode))}
+	}
+
+	var bidResponse openrtb.BidResponse
+
+	var adheseBidResponseArray []AdheseBid
+	if err := json.Unmarshal(response.Body, &adheseBidResponseArray); err != nil {
+		return nil, []error{err, WrapError(fmt.Sprintf("Response %v could not be parsed as generic Adhese bid.", string(response.Body)))}
+	}
+
+	var adheseBid = adheseBidResponseArray[0]
+
+	if adheseBid.Origin == "JERLICIA" {
```

Comment:
Can you please share some context on why this one publisher is treated specially?

Reply:
This is our own internal legacy system.

Reply:
Gotcha. Thank you for clarifying. Just wanted to make sure we're not hardcoding specific publishers.